### DISCUSSION TITLE
Issue with JSON validation in Markdown code blocks

### DISCUSSION THREAD

**Discussion by tpwo** [community/community#23409] [New to GitHub]
<p>Hi,</p>
<p>I’m not sure if this is the best place to report it.</p>
<p>But I think that I found an issue with JSON validation in Markdown files. The specific JSON code block seems to cause an issue with the validator, and in consequence, the rest of the Markdown file is marked an invalid, which seems like an unexpected behavior.</p>
<p>The issue is visible in blame view and when referring to the source code in issues and pull requests.</p>
<p>The very example of this behavior that I found:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/5/a5971932ab700408c15875ff3d7ad44fda52debe.png" data-download-href="/uploads/short-url/nCSvAzWUt8J0RxOWjf92zL8Nq6a.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/49607727/181076783-ada0327c-8502-4257-a14e-fc01c3f0e624.png" alt="image" data-base62-sha1="nCSvAzWUt8J0RxOWjf92zL8Nq6a" width="333" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/5/a5971932ab700408c15875ff3d7ad44fda52debe_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">540×810 13.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Direct link to the blame of the problematic file:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/pegn/spec/blame/2ad10e2c1bcb7230d39f1489ac983d21e4637573/doc/README.md#L150">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/pegn/spec/blame/2ad10e2c1bcb7230d39f1489ac983d21e4637573/doc/README.md#L150" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/49607727/181076784-c4c742d4-fa86-4e42-8420-ad0ec1b68cd6.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/5/25968f1fb41cb32d6badb4264eb945a88978c132_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/5/25968f1fb41cb32d6badb4264eb945a88978c132_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/5/25968f1fb41cb32d6badb4264eb945a88978c132.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/5/25968f1fb41cb32d6badb4264eb945a88978c132_2_10x10.png"></div>

<h3><a href="https://github.com/pegn/spec/blame/2ad10e2c1bcb7230d39f1489ac983d21e4637573/doc/README.md#L150" target="_blank" rel="noopener nofollow ugc">pegn/spec</a></h3>

  <p>PEGN is a universal notation for expressing any grammar without any specific implementation in mind. - pegn/spec</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Best regards,<br>
Tomasz</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The JSON isn’t terminated properly, and <code>...</code> isn’t valid JSON syntax. I wonder if syntax highlighting would work correctly if you close the open JSON blocks after the ellipsis? But I agree that highlighting the rest of the file beyond the code block like invalid JSON is a bug. :sweat_smile:</p>
<p>You can try reporting the problem to the GitHub feedback form: <a href="https://support.github.com/contact/feedback" class="inline-onebox">Share feedback - GitHub Support</a></p>

**Comment by tpwo** [comment]
<p>I tested on a toy example, and it seems that you’re right. Closing opened blocks makes this bug disappear:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/a/caa52bd4220167b0db2b2779cf158614baeffca0.png" data-download-href="/uploads/short-url/sUGkfRmmuO5z3vhhUQ9ZeCPd864.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/49607727/181076788-660199d7-3e84-4b2d-a42e-050f5f48a3bd.png" alt="image" data-base62-sha1="sUGkfRmmuO5z3vhhUQ9ZeCPd864" width="283" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/a/caa52bd4220167b0db2b2779cf158614baeffca0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">449×793 10.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks for the link, I’ll report it :smiley:</p>

**Comment by tpwo** [comment]
<p>I noticed that just skipping the closing tags makes this issue to appear:</p>
<p><img src="https://user-images.githubusercontent.com/49607727/181076791-bf4d5d4a-a2d2-4472-a65b-f35fe4f52eae.png" alt="image" data-base62-sha1="qRpr5LsAdbYP6RjJYgKeXqDwjOj" width="399" height="317"></p>
